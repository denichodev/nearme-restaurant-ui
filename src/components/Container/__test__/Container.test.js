import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { Container } from '../index';

describe('<Container />', () => {
  it('renders correctly', () => {
    const props = {
      location: {
        pathname: '/',
      },
    };

    const wrapper = shallow(<Container {...props} />)

    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('hides footer if the page is detail page', () => {
    const props = {
      location: {
        pathname: '/r/something',
      },
    };

    const wrapper = shallow(<Container {...props} />)

    expect(wrapper.find('Footer')).toHaveLength(0);
  })
});
