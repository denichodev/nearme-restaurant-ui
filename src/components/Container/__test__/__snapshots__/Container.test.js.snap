// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> renders correctly 1`] = `
<div
  id="container"
>
  <div
    className="headerNav"
  >
    <Grid
      className=""
      fluid={false}
    >
      <Row
        className="headerWrapper"
      >
        <div
          className="logo"
        >
          <Link
            replace={false}
            to="/"
          >
            <i
              className="fas fa-utensils"
            />
          </Link>
        </div>
        <nav
          className="desktopNav"
        >
          <ul>
            <li>
              <Link
                replace={false}
                to="/explore"
              >
                Explore
              </Link>
            </li>
            <li>
              <Link
                replace={false}
                to="/me"
              >
                Login
              </Link>
            </li>
          </ul>
        </nav>
      </Row>
    </Grid>
  </div>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      path="/r/:slug"
    />
    <Route
      component={[Function]}
    />
  </Switch>
  <Footer />
</div>
`;
